### PR TITLE
ci: add a new release workflow

### PR DESCRIPTION
**Pull request by andrii-zakurenyi**
Updates to GitHub workflows:
- Removed both `test-unit` and `test-sanity` workflows.
- Created a new `test` workflow which uses `mage` to init, test and build coverage report.
- Created a new `release` workflow which uses `mage` to run sanity check, then build and publish the collection.

Updates to Magefile:
- Delete all namespaces (Ansible, Venv, Py, Job).
- Reduce number of targets

Output of `mage -l` command:

```text
Targets:
  build         📦 Build packages the collection into a publishable archive.
  bump          🔼 Bump increments version in the galaxy file of the collection, using yq.
  changelog     📜 Changelog helps with creating a release changelog.
  clean         🧹 Clean removes '.artifact/', '.cache/', 'tests/output/' directories from the project.
  doctor        🔍 Doctor will validate the required tools and environment variables are available.
  init          ✨ Init unfolds initial environment for productive work.
  initCI        🎩 InitCI initializes a new Python virtual environment with given version of Ansible installed.
  publish       🚀 Publish sends archived collection to Ansible Galaxy.
  test          🧪 Test runs unit and sanity tests.
  testSanity    🧪 TestSanity runs sanity tests in containers.
  testUnit      🧪 TestUnit runs unit tests in containers.
```

[AB#500727](https://thycotic.visualstudio.com/4a89362e-1361-424f-a291-a8f57c2a8991/_workitems/edit/500727)

### CODE REVIEW

**Comment by andrii-zakurenyi** [comment]
Less than a hundred lines of logs for that "build-and-publish" step. Charming :sparkles:

**Comment by andrii-zakurenyi** [comment]
Not sure what should be the trigger of this workflow though

**Comment by andrii-zakurenyi** [comment]
Will it be good enough? :man_shrugging: 

```yaml
on:
  push:
    tags:
      - v*
```

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -1,18 +1,43 @@
 ---
 name: release
 on:
+  pull_request:
   workflow_dispatch:
+  push:
```
Comment (tdillenbeck) [inline .github/workflows/release.yaml:6]: I don't know all the context, but this seems right to me.

**Review by tdillenbeck — APPROVED** [review]

**Review by sheldonhull — CHANGES_REQUESTED** [review]
The release command should be in Mage. It could run the release plus create the github release commands which actually would tag. Let me look today and I’ll provide review feedback. You won’t use pull request  on trigger.

**Comment by sheldonhull** [comment]
@andrii-zakurenyi included some other tasks to help with this, as ideally you can run locally the same as remote. These changes include:

- install just a single venv for running publishing commands.
- using that venv (not super important in CI, but ideal for local). 
- using semver bump and `yq` bump the galaxy yaml version as part of development.


I'm ok merging this as is now to break things into smaller pieces, but there's one last thing to look at afterward... does `ansible-galaxy` publishing actually create a github release? If not then we can either look at using the github action that does the build and workflow creation, OR we add a github release via `gh release create` invocation to the mage release process that uses the semver bump. `gh release` actually adds the tag for us so we don't push manual tags anymore. Makes it nice for the release logs. 

Almost no github repos get manual tags due to this github release tag creation process.

**Comment by sheldonhull** [comment]
flipped to draft as `do-not-merge` is not a blocking condition. I'll catch up next week with you and if you want to pair on anything to wrap it up just let me know. Cheers

**Comment by sheldonhull** [comment]
Haven't forgotten about this. Will review soon as lots of changes and want to observe the improvements you made.  Cheers

**Comment by sheldonhull** [comment]
Merging. Will handle further review or changes for docs in a different PR. Thank you for this hard work @andrii-zakurenyi

**Pull request merged**